ruby/src/IceRuby/Communicator.cpp

```diff
@@ -40,128 +40,57 @@ static const rb_data_type_t IceRuby_CommunicatorType = {
     .flags = RUBY_TYPED_FREE_IMMEDIATELY,
 };
 
-namespace
-{
-    class CommunicatorDestroyer
-    {
-    public:
-        CommunicatorDestroyer(const Ice::CommunicatorPtr& c) : _communicator(c) {}
-
-        ~CommunicatorDestroyer() { _communicator->destroy(); }
-
-    private:
-        Ice::CommunicatorPtr _communicator;
-    };
-}
-
 extern "C" VALUE
 IceRuby_initialize(int argc, VALUE* argv, VALUE /*self*/)
 {
+    // This is the implementation of the Ice.initialize module method. There is no Ruby wrapper.
```

Comment:
With a Ruby wrapper, we would pass only initData, and the wrapper would create Properties from args when needed.